inginious/frontend/pages/course_admin/api_tokens.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+from collections import OrderedDict
+
+import flask
+
+import jwt
+from datetime import datetime
+
+from inginious.frontend.pages.course_admin.utils import INGIniousAdminPage
+
+class CourseAPITokens(INGIniousAdminPage):
+    """ List information about api tokens """
+
+    def GET_AUTH(self, courseid):  # pylint: disable=arguments-differ
+        """ GET request """
+        course, __ = self.get_course_and_check_rights(courseid)
+        return self.page(course)
+
+    def POST_AUTH(self, courseid):
+        """ POST request """
+        course, __ = self.get_course_and_check_rights(courseid)
+        user_input = flask.request.form.copy()
+        username = self.user_manager.session_username()
+
+        if "add_token" in user_input:
+            descr = user_input.get("descr", "")
+            expire = user_input.get("expiration", "")
+            if expire == "":
+                expire = "no"
```

Comment:
Do not store `no` in database. 

Either use a separate boolean, set an pre-defined max date (we may have already use this somewhere in the code, that would equals to something like year 9999), or rely on the mutable specs of a Mongo document and don't store any expiration if it should not expire.